### PR TITLE
Default creators value should be unicode

### PR DESCRIPTION
**Pull request by lentinj**
(repeat of #93, this time without broken history)

Stumbled on this whilst tinkering with transmogrify.dexterity - if no value is supplied for a field, the schemaupdater tries to get the default value for a field then set it to that, similar to the provided unit test. If the default value for creators isn't unicode, then it fails with a validation error.

I'm not entirely sure why schemaupdater does this mind, the code seems a bit odd. But ensuring the default value passes validation seems reasonable, if somewhat academic.

### CODE REVIEW

**Comment by jensens** [comment]
@lentinj Any interest to bring this further?

**Comment by lentinj** [comment]
*wakes up with a start*

@davisagli true, but arbitarily assuming that the userid is UTF8-encoded bytes smells like a bodge.

Currently, the userid is an ASCIILine, at least in the registration form. In theory one shouldn't have any non-ASCII characters in their userid:-
https://github.com/plone/plone.app.users/blob/master/plone/app/users/schema.py#L86
https://github.com/plone/plone.app.users/blob/1.1.x/plone/app/users/browser/register.py#L46
...so if we're failing early, failing often I should assume ASCII too. The real bug here is plone.app.users and plone.app.dexterity's schemas don't match, but changing them sounds scary.

Removing this restriction on the registration form would be nice. However, I can imagine the assumption that the userid is ASCII runs deep.

(apparently I'm in a pedantic mood today, apologies. Stop whining and get on with it could be a valid response.)

**Comment by davisagli** [comment]
Okay, that's a fair argument.

File: plone/app/dexterity/behaviors/metadata.py [hunk 1]
```diff
@@ -198,7 +198,7 @@ class IOwnership(model.Schema):
 # make sure the add form shows the default creator
 def creatorsDefault(data):
     user = getSecurityManager().getUser()
-    return user and (user.getId(),)
+    return user and (unicode(user.getId()),)
```
Comment (pbauer) [inline plone/app/dexterity/behaviors/metadata.py]: better safe_unicode than sorry (from Products.CMFPlone.utils import safe_unicode)

Comment (davisagli) [inline plone/app/dexterity/behaviors/metadata.py]: Read the PR discussion -- it's intentionally not assuming utf8 encoding.

**Comment by jensens** [comment]
So the encoding of a userid is ASCII? Really? Is the by intend or by accident? What if the userid comes from i.e. LDAP, some SQL-DB or OpenID/OAuth? 

Ok, the PR is the wrong place to discuss it, but the whole thing smells. We're discussing symptoms of a much deeper problem I guess. At least at this place only a workaround is possible, not a solution.

**Comment by davisagli** [comment]
Oh yeah, that does remind me, I've seen utf8-encoded user ids coming from openldap in the wild.

**Comment by jensens** [comment]
Sure. And IMO there is nothing wrong with it.

**Comment by lentinj** [comment]
Okay, so the other end of this assumption is here I think, so happy to go safe_unicode:-

https://github.com/plone/plone.dexterity/blob/master/plone/dexterity/content.py#L420

**Comment by jensens** [comment]
looks imo good, may you rebase so we can merge?

**Comment by lentinj** [comment]
Done.

**Pull request merged**